File: app/src/java/java/com/example/ama/viewer/data/repo/ApiRepositoryImpl.java
package com.example.ama.viewer.data.repo;

import com.example.ama.viewer.data.api.GithubApi;
import com.example.ama.viewer.data.api.dto.GithubUserDTO;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;
import io.reactivex.schedulers.Schedulers;

@Singleton
public class ApiRepositoryImpl implements ApiRepository {
    private GithubApi githubApi;

    @Inject
    public ApiRepositoryImpl(GithubApi githubApi) {
        this.githubApi = githubApi;
    }

    @Override
    public Observable<GithubUserDTO> loadData(String login) {
        return githubApi.getUserByUsername(login)
                .subscribeOn(Schedulers.io());
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/data/repo/ApiRepositoryImpl.kt
package com.example.ama.viewer.data.repo

import com.example.ama.viewer.data.api.GithubApi
import com.example.ama.viewer.data.api.dto.GithubUserDTO
import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class ApiRepositoryImpl @Inject constructor(
        private val githubApi: GithubApi
) : ApiRepository {

    override fun loadData(login: String): Observable<GithubUserDTO> =
            githubApi.getUserByUsername(login)
                    .subscribeOn(Schedulers.io())
}